### PR TITLE
Feat/dev 845 urls metadata helper functions

### PR DESCRIPTION
**Pull request by XukunXiangUChicago**
Add helper methods for `Document` class to get `url` or `state` of a requested url_type

### CODE REVIEW

File: indexclient/client.py [hunk 1]
```diff
@@ -415,6 +415,12 @@ class DocumentDeletedError(Exception):
     pass
 
 
+class UrlMetadata(NamedTuple):
```
Comment (mpsolano) [inline indexclient/client.py]: I think typing.NamedTuple is only available in Python3.6+

Comment (BenX00) [reply indexclient/client.py]: It's available for 3.5 as well but you do have to use [a special syntax](https://docs.python.org/3/library/typing.html#typing.NamedTuple) for creating the new class. It's rather moot tho cuz 2.7 only supports the [namedtuple](https://docs.python.org/2.7/library/collections.html#collections.namedtuple) from the collections module.

File: indexclient/client.py [hunk 2]
```diff
@@ -526,6 +532,25 @@ def delete(self):
                             params={"rev": self.rev})
         self._deleted = True
 
+    def get_url_metadata_by_type(self, request_prop: str, url_type: str) -> Optional[str]:
```
Comment (kulgan) [inline indexclient/client.py]: I think having this method return the actual `URLMetadata` will more useful in the long run, you can then add explicit methods that return the state and the url named according the definitions in the ticket. It also makes the `URLMetadata` data structure much more useful, though I think a more appropriate definition of this data structure will be using `TypedDict`

File: indexclient/client.py [hunk 3]
```diff
@@ -526,6 +529,39 @@ def delete(self):
                             params={"rev": self.rev})
         self._deleted = True
 
+    def get_url_metadata_by_type(self, url_type):
```
Comment (kulgan) [inline indexclient/client.py:532]: Some google python docstring will be nice

**Review by kulgan — APPROVED** [review]

**Review by jiakf — APPROVED** [review]

**Review by ProfOak — APPROVED** [review]
Thanks for taking care of this. It was much needed. :+1:

**Pull request merged**